lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -418,7 +418,7 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     // FIXME: Need to communicate more information on the process exit status.
     getDelegate().commandProcessFinished(context.job.getForCommand(), handle,
                                          status);
-    return (status == 0);
+    return (status == 0) || (WIFSIGNALED(status) && (WTERMSIG(status) == SIGINT || WTERMSIG(status) == SIGKILL));
```

Comment:
I think we should go ahead and clean this up to report slightly richer status to commandProcessFinished -- we should try and extract our idea of whether or not the process was cancelled vs failed or killed and then pass that as information both to the callback and then used to derive our result here.